clientconn.go

```diff
@@ -934,10 +933,14 @@ func equalAddresses(a, b []resolver.Address) bool {
 // updateAddrs updates ac.addrs with the new addresses list and handles active
 // connections or connection attempts.
 func (ac *addrConn) updateAddrs(addrs []resolver.Address) {
-	ac.mu.Lock()
-	channelz.Infof(logger, ac.channelz, "addrConn: updateAddrs curAddr: %v, addrs: %v", pretty.ToJSON(ac.curAddr), pretty.ToJSON(addrs))
-
 	addrs = copyAddressesWithoutBalancerAttributes(addrs)
+	limit := len(addrs)
+	if limit > 5 {
+		limit = 5
+	}
+	channelz.Infof(logger, ac.channelz, "addrConn: updateAddrs addrs (%d of %d): %v", limit, len(addrs), addrs[:limit])
```

Comment:
Can you show what this looks like, please?  (Mainly I'm not sure: does `%v` on a slice call `String()` for each member?)

Reply:
Looks something like this:
```
addrConn: updateAddrs addrs (3 of 3): [{Addr: "127.0.0.1:35771", ServerName: "", } {Addr: "127.0.0.1:41729", ServerName: "", } {Addr: "127.0.0.1:36587", ServerName: "", }]
```

Reply:
Great, thanks!